File: java/simpleserver/src/main/java/com/rc/simpleserver/controllers/EchoController.java
package com.rc.simpleserver.controllers;

import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class EchoController {

    @GetMapping(value = "/echo", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
    public @ResponseBody String doEcho(@RequestBody String request) {
        return request;
    }
}

File: kotlin/simpleserver/src/main/kotlin/com/rc/simpleserver/controllers/EchoController.kt
package com.rc.simpleserver.controllers

import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.RestController

@RestController
class EchoController {

    @GetMapping(value = ["/echo"], consumes = [ MediaType.APPLICATION_JSON_VALUE ])
    @ResponseBody fun doEcho(@RequestBody request: String): String {
        return request
    }
}
